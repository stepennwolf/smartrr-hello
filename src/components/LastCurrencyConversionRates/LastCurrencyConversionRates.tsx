export function LastCurrencyConversionRates({ conversionRates, currencies}: LastCurrencyConversionRatesProps) {
  if (conversionRates.length < 1) {
      return <p>Currency conversion rates haven't been provided yet</p>
  }

  return (
      <section className="LastCurrencyConversionRates">
        <h2>Last conversion rates for {currencies}</h2>
          <ul>
              {conversionRates.map(({ rate, date }) => {
                const formattedDate = new Intl.DateTimeFormat('en-US', {
                  year: 'numeric',
                  month: 'numeric',
                  day: 'numeric',
                  hour: 'numeric',
                  minute: 'numeric',
                  second: 'numeric',
                }).format(date);

                return <li key={formattedDate}>{formattedDate} - <strong>{rate}</strong></li>
              })}
          </ul>
      </section>
    )
}


export type LastCurrencyConversionRatesProps = {
    currencies: string; // <FROM>/<TO> format
    conversionRates: ConversionRatesProp[];
}

export type ConversionRatesProp = {
    rate: number;
    date: Date;
}

export default LastCurrencyConversionRates;
