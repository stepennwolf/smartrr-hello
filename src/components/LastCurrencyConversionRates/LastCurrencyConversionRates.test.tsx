import renderer from 'react-test-renderer';
import LastCurrencyConversionRates from './LastCurrencyConversionRates';

describe('<LastCurrencyConversionRates/>', () => {
  test('renders with empty data', () => {
    const testProps = {
      currencies: 'USD/BRL',
      conversionRates: []
    };

    const tree = renderer
      .create(<LastCurrencyConversionRates {...testProps} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });


  test('renders with supplied data', () => {
    const testProps = {
      currencies: 'USD/BRL',
      conversionRates: [
        {rate: 5.0123, date: new Date(1675786531551)},
        {rate: 7.0453, date: new Date(1675793731551)},
        {rate: 4.1432, date: new Date(1675800931551)},
      ]
    };

    const tree = renderer
      .create(<LastCurrencyConversionRates {...testProps} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
