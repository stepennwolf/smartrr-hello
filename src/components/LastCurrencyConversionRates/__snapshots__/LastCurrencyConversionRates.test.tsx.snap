// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LastCurrencyConversionRates/> renders with empty data 1`] = `
<p>
  Currency conversion rates haven't been provided yet
</p>
`;

exports[`<LastCurrencyConversionRates/> renders with supplied data 1`] = `
<section
  className="LastCurrencyConversionRates"
>
  <h2>
    Last conversion rates for 
    USD/BRL
  </h2>
  <ul>
    <li>
      2/7/2023, 6:15:31 PM
       - 
      <strong>
        5.0123
      </strong>
    </li>
    <li>
      2/7/2023, 8:15:31 PM
       - 
      <strong>
        7.0453
      </strong>
    </li>
    <li>
      2/7/2023, 10:15:31 PM
       - 
      <strong>
        4.1432
      </strong>
    </li>
  </ul>
</section>
`;
